internal/tls/tls.go

```diff
@@ -169,6 +169,12 @@ func generateCertificate(certFile, keyFile string) (*tls.Config, error) {
 		return nil, errors.WithStack(err)
 	}
 
+	if _, err := os.Stat(filepath.Dir(certFile)); errors.Is(err, fs.ErrNotExist) {
```

Comment:
In theory, the key file might be in a different directory so this check should be repeated.

Reply:
totally. overlooked that in v2 we could have separate paths as opposed to the v1 base dir for TLS.